raiden/transfer/mediated_transfer/events.py

```diff
@@ -292,12 +293,25 @@ def __init__(
         recipient: Address,
         channel_identifier: ChannelID,
         message_identifier: MessageID,
-        signed_locked_transfer: LockedTransferUnsignedState
+        signed_locked_transfer: LockedTransfer
     ) -> None:
         super().__init__(recipient, channel_identifier, message_identifier)
 
         self.signed_locked_transfer = signed_locked_transfer
 
+        canonical_identifier = CanonicalIdentifier(
+            channel_identifier=channel_identifier,
+            token_network_address=self.signed_locked_transfer.token_network_address,
+            chain_identifier=self.signed_locked_transfer.chain_id
+        )
+        self.balance_proof = BalanceProofUnsignedState(
```

Comment:
i find this a bit confusing: why is this `Unsigned` if it uses a `signed_locked_transfer`?

Reply:
I changed the signed_locked_transfer type to LockedTransfer, as that was what it actually received from the caller. LockedTransfer doesn't have enought information to create a signed balance proof. I think we could change signed_locked_transfer's name to locked_transfer. in order to prevent confusions.

Reply:
@mauricioirace was the type hint in the signature for this function wrong?

Reply:
Yes, both worked. But this one is more more correct, as the locked transfer and the bp are signed.